affinityworks/main
# resolve "BUG: existing user joining group with oauth broken (#657)"

resolves #657 
________________

# Functionality

Handle OAuth group-joining flow corner cases by checking in "step 1" to see if a person with email matching OAuth email already exists, then:

* if person does not yet exist, handle as before
* if they do exist:
  * if they are already member of group: log them in and redirect them to group dashboard
  * if they aren't member and lack contact info: redirect to members/edit` form where we gather their contact info, make them a member, and record their oauth info (but not until then)
  * if they aren't member and have all contact info: don't bother redirecting. just make them a member, record their token, and log them into group dashboard page

Handle email group-joining flow corner cases as follows:

* if user follows email join path and submits form with already-existing email, redirect them to login page with error message
* after they login, redirect them back to group join page, where they may join with one click

# Manual Testing Notes
(see forthcoming comment. it's hard to create the state necessary to test all these corner cases)
